mi/dataset/parser/glider.py

```diff
@@ -1042,9 +1044,22 @@ def _has_science_data(data_dict, particle_class):
         """
         Examine the data_dict to see if it contains particle parameters
         """
-        for key, value in data_dict.iteritems():
-            if not(isnan(float(value))) and key in particle_class.science_parameters:
-                return True
+        return_value = False
+
+        # Modified to make this check more efficient
+        if len(particle_class.science_parameters) < len(data_dict):
+            for key in particle_class.science_parameters:
+                value = data_dict.get(key)
```

Comment:
if the value is 0, this will result in a false positive, maybe `value = data_dict.get(key, None)` and then `if value is not None and not(isnan(float(value)))`

Reply:
Good catch. Will fix